docs/java-rest/high-level/apis/_index.asciidoc

```diff
@@ -104,10 +104,28 @@ include-tagged::{doc-tests}/CRUDDocumentationIT.java[index-execute]
 [[java-rest-high-document-index-async]]
 ==== Asynchronous Execution
 
+The asynchronous execution of an index request requires both the `IndexRequest`
```

Comment:
Note: I used the same wording in all docs, so feel free to only comment once. I'll apply any potential changes by grepping and sedding all the files together.